rotkehlchen/chain/ethereum/modules/liquity/trove.py

```diff
@@ -202,23 +202,30 @@ def __init__(
             )
             raise ModuleInitializationFailure('Liquity Subgraph remote error') from e
 
-    def get_positions(
-        self,
-        addresses_list: List[ChecksumEvmAddress],
-    ) -> Dict[ChecksumEvmAddress, Trove]:
-        contract = EvmContract(
+        self.trove_manager_contract = EvmContract(
             address=LIQUITY_TROVE_MANAGER.address,
             abi=LIQUITY_TROVE_MANAGER.abi,
             deployed_block=LIQUITY_TROVE_MANAGER.deployed_block,
         )
+        self.stability_pool_contract = EvmContract(
+            address=LIQUITY_STABILITY_POOL.address,
+            abi=LIQUITY_STABILITY_POOL.abi,
+            deployed_block=LIQUITY_STABILITY_POOL.deployed_block,
+        )
+
+    def get_positions(
+        self,
+        addresses_list: List[ChecksumEvmAddress],
+    ) -> Dict[ChecksumEvmAddress, Trove]:
+        """Query liquity contract to detect open troves"""
         # make a copy of the list to avoid modifications in the list that is passed as argument
-        addresses = list(addresses_list)
+        addresses = addresses_list.copy()
```

Comment:
why are you copying the list here?

Reply:
Because later I do `addresses += proxied_addresses.values()` and since you are passing the list as an argument you have to prevent from modifying it